includes/class-wc-stripe-helper.php

```diff
@@ -585,17 +585,29 @@ public static function should_enqueue_in_current_tab_section( $tab, $section ) {
 	}
 
 	public static function should_load_scripts_on_product_page() {
-		$prb_locations = self::get_settings( null, 'payment_request_button_locations' ) ?? [ 'product', 'cart' ];
-		if ( ! in_array( 'product', $prb_locations, true ) ) {
+		$are_prbs_enabled = self::get_settings( null, 'payment_request' ) ?? 'yes';
+		$prb_locations    = self::get_settings( null, 'payment_request_button_locations' ) ?? [ 'product', 'cart' ];
+
+		// We run the filter when 1 of the following is true:
+		//   1. The PRBs are disabled; or
+		//   2. The PRB location settings are emptied and saved in the GUI (results in a non-array value).
+		//   3. The PRBs are disabled on product pages.
+		if ( 'yes' !== $are_prbs_enabled || ! is_array( $prb_locations ) || ! in_array( 'product', $prb_locations, true ) ) {
 			return apply_filters( 'wc_stripe_load_scripts_on_product_page_when_prbs_disabled', true );
 		}
 
 		return true;
 	}
 
 	public static function should_load_scripts_on_cart_page() {
-		$prb_locations = self::get_settings( null, 'payment_request_button_locations' ) ?? [ 'product', 'cart' ];
-		if ( ! in_array( 'cart', $prb_locations, true ) ) {
+		$are_prbs_enabled = self::get_settings( null, 'payment_request' ) ?? 'yes';
+		$prb_locations    = self::get_settings( null, 'payment_request_button_locations' ) ?? [ 'product', 'cart' ];
+
+		// We run the filter when 1 of the following is true:
+		//   1. The PRBs are disabled; or
+		//   2. The PRB location settings are emptied and saved in the GUI (results in a non-array value).
+		//   3. The PRBs are disabled on the cart page.
+		if ( 'yes' !== $are_prbs_enabled || ! is_array( $prb_locations ) || ! in_array( 'cart', $prb_locations, true ) ) {
```

Comment:
This is admittedly a little pedantic, but given that this conditional is exactly the same besides the value we're looking for in the array, what do you think about making a new (_groan_) static method, something like:

```php
public static function should_load_scripts_for_prb_location( $location ) {
        // maybe not exactly this, but _some_ sanity check on $location
        $location         = in_array( $location, [ 'product', 'cart' ] ) ? $location : 'product';
        $are_prbs_enabled = self::get_settings( null, 'payment_request' ) ?? 'yes';
		$prb_locations    = self::get_settings( null, 'payment_request_button_locations' ) ?? [ 'product', 'cart' ];

		// We run the filter when 1 of the following is true:
		//   1. The PRBs are disabled; or
		//   2. The PRB location settings are emptied and saved in the GUI (results in a non-array value).
		//   3. The PRBs are disabled on the cart page.
		return ( 'yes' !== $are_prbs_enabled || ! is_array( $prb_locations ) || ! in_array( $location, $prb_locations, true ) );
}
```

Then the conditional in both methods would just be:

```php
if ( self::should_load_scripts_for_prb_location( 'product' ) ) {
```
or

```php
if ( self::should_load_scripts_for_prb_location( 'cart' ) ) {
```

Reply:
> what do you think about making a new (groan) static method

😄 

I'll get on it!

Reply:
Oh and probably `private` instead of `public` on the new static method, so we keep the "public" API limited to the product/cart specific methods. The new method I propose would just be an internal implementation detail.

Reply:
Fixed in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2179/commits/f93fb2611070b5ca1aedf9acba71bf92d8523dfd 🎉